### ISSUE TITLE
Replicate SQuAD results 

### ISSUE THREAD

**Issue by zaidalyafeai** [aub-mind/arabert#30]
Hey, I am trying to replicate the SQuAD results for arabert v0.1, 

First, I combined Arabic-SQuAD and arcd-train then used the preprocessing script 

```
!python arcd_preprocessing.py \
    --input_file="SOQAL/data/arcd-test.json" \
    --output_file="arcd-test-pre.json" \
    --do_farasa_tokenization=False \
    --use_farasapy=False

!python arcd_preprocessing.py \
    --input_file="turk_combined_all.json" \
    --output_file="turk_combined_all_pre.json" \
    --do_farasa_tokenization=False \
    --use_farasapy=False
```
And used the following for training 

```
!python run_squad.py \
  --vocab_file=$BUCKET_NAME/arabert/vocab.txt \
  --bert_config_file=$BUCKET_NAME/arabert/config.json \
  --init_checkpoint=$BUCKET_NAME/arabert/arabert01_model.ckpt \
  --do_train=True \
  --train_file=turk_combined_all_pre.json \
  --do_predict=True \
  --predict_file=arcd-test-pre.json \
  --train_batch_size=32 \
  --predict_batch_size=24 \
  --learning_rate=3e-5 \
  --num_train_epochs=4 \
  --do_lower_case=False \
  --use_tpu=True \
  --tpu_name=$TPU_ADDRESS \
  --max_seq_length=384 \
  --doc_stride=128 \
  --output_dir=$BUCKET_NAME/$MODEL_NAME
```

After evaluation using the script 

``` 
!python3 SOQAL/bert/evaluate.py arcd-test-pre.json ../bucket/squad_arabert/predictions.json
```

I get 
```{"exact_match": 30.34188034188034, "f1": 65.53764467213384, "exact_sentence": 91.31054131054131}```

This is lower than the reported results, what am doing wrong ?

**Comment by WissamAntoun** [comment]
The only difference can see is with the evaluation script.

in my script I deleted the remove Arabic articles function from the normalization func

```
def normalize_answer(s):
    """Lower text and remove punctuation, articles and extra whitespace."""
    def remove_articles(text):
        return re.sub(r'\b(a|an|the)\b', ' ', text)

    def white_space_fix(text):
        return ' '.join(text.split())

    def remove_punc(text):
        exclude = set(string.punctuation)
        return ''.join(ch for ch in text if ch not in exclude)

    def lower(text):
        return text.lower()

    return white_space_fix(remove_articles(remove_punc(lower(s))))
```

not sure if this is the reason.

can you try with arabertv1 and see if the problem persists?
I don't think its an issue with the hyperparameters since the results are far lower than expected

**Comment by zaidalyafeai** [comment]
Removed `remove_articles_ar` and the problem still persists. Also I fine-tuned arabertv1 and I get similar results 

```
{"exact_match": 31.48148148148148, "f1": 67.18799136378952, "exact_sentence": 90.5982905982906}
```
